mixer/adapter/rbac/rbacStore.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"strings"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/authorization"
+)
+
+// Rbac interface
+type Rbac interface {
+	CheckPermission(inst *authorization.Instance, env adapter.Env) (bool, error)
+}
+
+// Information about a ServiceRole and associted ServiceRoleBindings
+type roleInfo struct {
+	// ServiceRole proto definition
+	info *rbacproto.ServiceRole
+
+	// A set of ServiceRoleBindings that refer to this role.
+	bindings map[string]*rbacproto.ServiceRoleBinding
+}
+
+// maps role name to role info
+type rolesByName map[string]*roleInfo
+
+// maps namespace to a set of roles in the namespace
+type rolesMapByNamespace map[string]rolesByName
+
+// configStore contains all ServiceRole and ServiceRoleBinding information.
+// configStore implements Rbac interface.
+type configStore struct {
+	// All the roles organized per namespace.
+	roles rolesMapByNamespace
+}
+
+// Create a RoleInfo object.
+func newRoleInfo(spec *rbacproto.ServiceRole) *roleInfo {
+	return &roleInfo{
+		info: spec,
+	}
+}
+
+// Set a binding for a given Service role.
+func (ri *roleInfo) setBinding(name string, spec *rbacproto.ServiceRoleBinding) {
+	if ri == nil {
+		return
+	}
+	if ri.bindings == nil {
+		ri.bindings = make(map[string]*rbacproto.ServiceRoleBinding)
+	}
+	ri.bindings[name] = spec
+}
+
+// Update roles in the RBAC store.
+func (rs *configStore) changeRoles(roles rolesMapByNamespace) {
+	rs.roles = roles
+}
+
+// CheckPermission checks permission for a given request. This is the main API called
+// by RBAC adapter at runtime to authorize requests.
+func (rs *configStore) CheckPermission(inst *authorization.Instance, env adapter.Env) (bool, error) {
+	namespace := inst.Action.Namespace
+	if namespace == "" {
+		env.Logger().Errorf("Missing namespace")
```

Comment:
is this an error condition?  If so, why do we return nil error? Should this be a warning? (I have the same question for the next 3 errorfs).

Reply:
I changed the code to return the errors.